extensions/registration-service/src/main/java/org/eclipse/dataspaceconnector/registration/credential/VerifiableCredentialServiceImpl.java

```diff
@@ -0,0 +1,99 @@
+import org.eclipse.dataspaceconnector.iam.did.spi.key.PrivateKeyWrapper;
+import org.eclipse.dataspaceconnector.iam.did.spi.resolution.DidResolverRegistry;
+import org.eclipse.dataspaceconnector.identityhub.client.IdentityHubClient;
+import org.eclipse.dataspaceconnector.identityhub.credentials.VerifiableCredentialsJwtService;
+import org.eclipse.dataspaceconnector.identityhub.credentials.model.VerifiableCredential;
+import org.eclipse.dataspaceconnector.registration.authority.model.Participant;
+import org.eclipse.dataspaceconnector.spi.monitor.Monitor;
+import org.eclipse.dataspaceconnector.spi.response.StatusResult;
+import org.eclipse.dataspaceconnector.spi.result.Result;
+
+import java.util.Map;
+import java.util.UUID;
+
+import static org.eclipse.dataspaceconnector.spi.response.ResponseStatus.ERROR_RETRY;
+import static org.eclipse.dataspaceconnector.spi.response.ResponseStatus.FATAL_ERROR;
+
+public class VerifiableCredentialServiceImpl implements VerifiableCredentialService {
+    private static final String IDENTITY_HUB_SERVICE_TYPE = "IdentityHub";
+
+    private final Monitor monitor;
+    private final VerifiableCredentialsJwtService jwtService;
+    private final PrivateKeyWrapper privateKeyWrapper;
+    private final String dataspaceDid;
+    private final DidResolverRegistry resolverRegistry;
+    private final IdentityHubClient identityHubClient;
+
+    public VerifiableCredentialServiceImpl(Monitor monitor, VerifiableCredentialsJwtService jwtService, PrivateKeyWrapper privateKeyWrapper, String dataspaceDid, DidResolverRegistry resolverRegistry, IdentityHubClient identityHubClient) {
+        this.monitor = monitor;
+        this.jwtService = jwtService;
+        this.privateKeyWrapper = privateKeyWrapper;
+        this.dataspaceDid = dataspaceDid;
+        this.resolverRegistry = resolverRegistry;
+        this.identityHubClient = identityHubClient;
+    }
+
+    @Override
+    public StatusResult<Void> publishVerifiableCredential(Participant participant) {
+        var vc = VerifiableCredential.Builder.newInstance()
+                .id(UUID.randomUUID().toString())
+                .credentialSubject(Map.of("memberOfDataspace", dataspaceDid))
+                .build();
+
+        var subject = participant.getDid();
+        SignedJWT jwt;
+        try {
+            jwt = jwtService.buildSignedJwt(vc, dataspaceDid, subject, privateKeyWrapper);
+        } catch (Exception e) {
+            return StatusResult.failure(FATAL_ERROR, e.toString());
+        }
+        monitor.info("Created dataspace membership VC for " + subject);
+
+        var did = participant.getDid();
+        var didDocument = resolverRegistry.resolve(did);
+        if (didDocument.failed()) {
+            return StatusResult.failure(ERROR_RETRY, "Failed to resolve DID " + did + ". " + didDocument.getFailureDetail());
+        }
+        var hubBaseUrl = getIdentityHubBaseUrl(didDocument.getContent());
```

Comment:
I would add more `monitor.debug` statements at each step to indicate what work is being done as this helps a lot when debugging issues. See https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/extensions/iam/decentralized-identity/identity-did-service/src/main/java/org/eclipse/dataspaceconnector/identity/DecentralizedIdentityService.java#L70 for an example.